### DISCUSSION TITLE
[Run issue]: Problem running demos

### DISCUSSION THREAD

**Discussion by jonahpeter** [topas-nbio/TOPAS-nBio#10] [Run-Problem] [answered]
### What was the issue?

Hi all, I get the following error when running `source rundemos.sh` from the `/Applications/TOPAS-nBio` directory

```
(base) dhcp-10-250-246-187:Scoring jonahpeter$ cd /Applications/TOPAS-nBio/
(base) dhcp-10-250-246-187:TOPAS-nBio jonahpeter$ source rundemos.sh
Topas quitting, unable to open parameter file:../../scorers/DNAScorerStepByStep.txt
--failed -- dna/CharltonDNA.txt
Topas quitting, unable to open parameter file:../../scorers/DNAScorerStepByStep.txt
--failed -- dna/LinearDNA.txt
Topas quitting, unable to open parameter file:../../scorers/DNAScorerStepByStep.txt
--failed -- dna/CircularPlasmid.txt
^C
```

best,
Jonah

### TOPAS-nBio category

Example

### Version

3.0

### What Operative System (OS) are you running on?

MacOS

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by icmeyer** [comment]
Hi again Jonah, 
So it looks like there are two different example scripts. One in the source code directory (wherever you cloned nBio to) one in the build and install directories (`/Applications/TOPAS/OpenTOPAS-build` from the current README`). Trying the one in the source code directory gives me the same output that you got (aka it's broken!), but the one in the installation directory looks something like this

```
#!/bin/csh -f
#
cd examples/TimeFeature
../../bin/topas BoxWithinBox.txt
../../bin/topas CameraRotateAndZoom.txt
../../bin/topas CylinderGrowingInPhi.txt
cd ../UCSFETF
../../bin/topas User_SOBP24_Viewer.txt
../../bin/topas User_Beamline_WC_Viewer.txt
cd ../SpecialComponents
../../bin/topas QuadAndDipoleMagnets.txt
../../bin/topas PurgingMagnet_move.txt
../../bin/topas MultiLeafCollimator_sequence.txt
../../bin/topas RangeModulator_CurrentModulatedBeam.txt
cd ../Nozzle
../../bin/topas ScatteringNozzle_run.txt
../../bin/topas ScanningTargetMovingHorizontal.txt
cd ../Patient
unzip Abdomen.zip
../../bin/topas ViewAbdomen.txt
rm -rf Abdomen
tar -zxf synthetic_lung.tar.bz2
../../bin/topas DoseTo4DCT.txt
rm -rf synthetic_lung
unzip DICOM_Box
../../bin/topas Implant.txt
rm -rf DICOM_Box
cd ../TimeFeature
../../bin/topas Darkening.txt
rm *.csv
../../bin/topas Logo.txt
cd ../..
rm examples/*/NbParticlesInTime.txt
```

It should have the file extension `.csh`. You can try running that one for now while `rundemos.sh` is broken

**Reply by JoseRamosMendez** [reply, marked as answer]
The issue with the topas-nbio's `rundemos.sh` is the hierarchy of the subdirectories. That is, the script calls (e.g.,)

    topas dna/CharltonDNA.txt

But that parameter file calls for `../../DNAScorerStepByStep.txt` without entering the directory `examples/dna/`. But we recently moved to a continuous integration suite that you can download [here](https://github.com/topas-nbio/qi-topas-nbio). It does the same thing than `rundemos.sh` but in a more structured way.

> [!Note]
> Put attention while installing `nrtest` so you don't affect your current python installation

**Reply by jonahpeter** [reply]
Great, thank you both for the help!
